lib/rust/metamodel/src/java/graphviz.rs

```diff
@@ -0,0 +1,47 @@
+//! Generating graphical representations of Java type systems.
+
+use super::*;
+use crate::graphviz::EdgeType;
+use crate::graphviz::Graph;
+use crate::graphviz::Node;
+use crate::graphviz::NodeType;
+
+
+
+// =========================
+// === Graphviz Graphing ===
+// =========================
+
+/// Produce a graphviz graph of the datatypes.
+pub fn graph(java: &TypeGraph) -> Graph {
```

Comment:
For each language's `TypeGraph`, there's a function that produces a `crate::graphviz::Graph`. `crate::graphviz::Graph` has a display implementation.

Reply:
After thinking about it, I'm going to flip this interface around: For each `TypeGraph`, there will be an `impl From<TypeGraph> for graphviz::Graph`, so that all typegraphs can be graphed by one interface `graphviz::Graph::from(typegraph)` (and all `graphviz` functionality is accessed through the one `pub mod graphviz`).